### PR TITLE
[BUILD][STF] STF Refactoring

### PR DESCRIPTION
**Pull request by jafey**
I moved the STF-Parts (Framework, Tests, Self-Tests) to new packages stf and stf.test
@m273d15 added the gradle part, so that you can build the project with gradle.

### CODE REVIEW

**Comment by m273d15** [comment]
Did you executed the STF tests locally within your eclipse environment?
I was not able to start the STF eclipse instances. I always receive the following message if I want to run eclipse via the launch configurations:
```
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=256m; support was removed in 8.0
java.security.AccessControlException: access denied ("java.util.PropertyPermission" "eclipse.exitcode" "read")
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
	at java.security.AccessController.checkPermission(AccessController.java:884)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
	at java.lang.SecurityManager.checkPropertyAccess(SecurityManager.java:1294)
	at java.lang.System.getProperty(System.java:717)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1529)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1492)
Exception in thread "main" java.security.AccessControlException: access denied ("java.util.PropertyPermission" "osgi.noShutdown" "read")
	at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
	at java.security.AccessController.checkPermission(AccessController.java:884)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:549)
	at java.lang.SecurityManager.checkPropertyAccess(SecurityManager.java:1294)
	at java.lang.System.getProperty(System.java:717)
	at java.lang.Boolean.getBoolean(Boolean.java:254)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1501)
```

File: eclipse/test/resources/launch/Saros_STF_Alice.launch [hunk 1]
```diff
@@ -16,7 +16,7 @@
...
-<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Xmx512m -ea -XX:MaxPermSize=256m -Djava.security.manager &quot;-Djava.security.policy=file:${workspace_loc:saros.eclipse}/test/resources/stf/all.policy&quot;  -Dsaros.debug=true -Dsaros.testmode=12345 -Dsaros.sleepTime=200 -Djava.rmi.server.hostname=localhost -Dorg.eclipse.swtbot.keyboard.strategy=org.eclipse.swtbot.swt.finder.keyboard.MockKeyboardStrategy  -Dorg.eclipse.swtbot.keyboard.layout=saros.stf.server.bot.default -Duser.language=en -Duser.country=US -Dfile.encoding=UTF-8 -Djava.util.Arrays.useLegacyMergeSort=true"/>
+<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-Xmx512m -ea -XX:MaxPermSize=256m -Djava.security.manager &quot;-Djava.security.policy=file:${workspace_loc:saros.eclipse}/test/resources/all.policy&quot;  -Dsaros.debug=true -Dsaros.testmode=12345 -Dsaros.sleepTime=200 -Djava.rmi.server.hostname=localhost -Dorg.eclipse.swtbot.keyboard.strategy=org.eclipse.swtbot.swt.finder.keyboard.MockKeyboardStrategy  -Dorg.eclipse.swtbot.keyboard.layout=saros.stf.server.bot.default -Duser.language=en -Duser.country=US -Dfile.encoding=UTF-8 -Djava.util.Arrays.useLegacyMergeSort=true"/>
```
Comment (fzieris) [inline eclipse/test/resources/launch/Saros_STF_Alice.launch]: Regarding @m273d15's stacktrace: The relative path to the policy file should probably be changed, too.

Comment (jafey) [reply eclipse/test/resources/launch/Saros_STF_Alice.launch]: I fixed the configs and now I was able to run an STF-Test. @m273d15 please try again.

**Comment by m273d15** [comment]
I fixed a few small bugs and rebased the PR. The stf tests are successful on my local system.

**Pull request merged**